src/Settings.php

```diff
@@ -104,9 +101,54 @@ private function sanitize_option_key( $key = '' ) {
 	 * @return string
 	 */
 	public function get_options_prefix() {
+		if ( empty( $this->opts_prefix ) ) {
+			$this->set_options_prefix();
+		}
+
 		return $this->opts_prefix;
 	}
 
+	/**
+	 * Get an array of all of this extension's options without array keys having the redundant prefix.
+	 *
+	 * @return array
+	 */
+	public function get_all_options() {
+		$raw_options = $this->get_all_raw_options();
+
+		$result = [];
+
+		foreach ( $raw_options as $key => $value ) {
+			$abbr_key          = str_replace( $this->get_options_prefix(), '', $key );
```

Comment:
Minor, but, if we're going to call this method multiple times and the result will be the same on each iteration, we should move it out of the loop and capture the result in a variable, ie:

```php
$prefix = $this->get_options_prefix();

/* ...other code ... */

foreach ( $raw_options as $key => $value ) {
    $abbr_key = str_replace( $prefix, '', $key );
    /* ... other code ... */
}
```

Reply:
updated. good catch.

Reply:
@barryhughes ok to go?